src/crimson/osd/backfill_state.h

```diff
@@ -308,6 +308,37 @@ struct BackfillState::BackfillListener {
   virtual ~BackfillListener() = default;
 };
 
+// PeeringFacade -- a facade (in the GoF-defined meaning) simplifying
+// the interface of PeeringState. The motivation is to have an inventory
+// of behaviour that must be provided by a unit test's mock.
+struct BackfillState::PeeringFacade {
+  virtual hobject_t earliest_backfill() const = 0;
+  virtual const std::set<pg_shard_t>& get_backfill_targets() const = 0;
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/crimson/test_backfill.cc:188:24: error: conflicting return type specified for 'virtual std::set<pg_shard_t> BackfillFixture::PeeringFacade::get_backfill_targets() const'
  188 |   std::set<pg_shard_t> get_backfill_targets() const override {
      |                        ^~~~~~~~~~~~~~~~~~~~
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/crimson/test_backfill.cc:18:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/crimson/osd/backfill_state.h:316:39: note: overridden function is 'virtual const std::set<pg_shard_t>& crimson::osd::BackfillState::PeeringFacade::get_backfill_targets() const'
  316 |   virtual const std::set<pg_shard_t>& get_backfill_targets() const = 0;
      |                                       ^~~~~~~~~~~~~~~~~~~~
```

might need to update `src/test/crimson/test_backfill.cc` accordingly.

Reply:
Oops, updated.